### PR TITLE
Added minimum amount of time for consuming items.

### LINKED ISSUE #50701: Mustard has "0 seconds" drinking time
### Describe the bug

When drinking mustard, it has `0 seconds` consume time.

The game appears to calculate saiety and fullness on a clock tick, so this allows one to go from almost-dead starving to Engorged in zero seconds by repeatedly drinking just mustard.

### Steps To Reproduce

* Drink mustard
* Don't do anything else to cause the timer to tick, like moving or waiting or eating other foods
* Keep drinking that delicious yellow gold

### Expected behavior

* Player should get full and cannot drink any more.

### Screenshots

_No response_

### Versions and configuration

- OS: Linux
    - OS Version: Distributor ID: Ubuntu; Description: Ubuntu 20.04.2 LTS; Release: 20.04; Codename: focal; 
- Game Version: bc28921 [64-bit]
- Graphics Version: Tiles
- Game Language: System language [English]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    No Fungal Monsters [No_Fungi]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Gooseheaded**
#### Summary
Bugfixes "Characters with no mutations will now always take at least 1 second to consume items."

#### Purpose of change
Fixes #50701

As explained in #50701 , right now it is possible for characters to consume some items _instantly._ Assuming nothing out of the ordinary is occurring/there are no external factors (for example, mutations), consuming items in this manner is a strange experience. As mentioned in the issue above,
>[...] this allows one to go from almost-dead starving to Engorged in zero seconds by repeatedly drinking just mustard.

This problem arises from the fact that the `get_consume_time()` function uses `time_duration` for its calculations. Apparently, the `time_duration` type is only precise down to 1 second, meaning that any duration <= 1 second automatically gets rounded down to 0 seconds.

To verify, start a new game, and spawn a bottle of mustard. When you try to consume it, the UI will say it will take you "0 seconds" to do so. When you eat it, the in-game timer does not advance. If you repeatedly eat the mustard, the "Hunger" display in the sidebar will not update.

#### Describe the solution
I ensured that the minimum amount of time it takes for a character to consume an item is *1 second*. This applies to all foods, drinks, and drugs.

*PLEASE NOTE:* Some mutations can affect a character's consumption speed. In this PR, *mutations are deliberately immune to this restriction,* potentially allowing characters to eat instantly. I feel like that's appropriate; mutations are meant to give characters unnatural properties.

In other words, this change only applies to characters _with no mutations._

#### Describe alternatives you've considered

* Maybe increasing the precision in this calculation would be more _accurate_, but a 1 second consumption time sounds realistic to me, and was easier to implement.

#### Testing

I have added a new test that creates a character, gives it mustard, and calculates the time it would take for the character to consume it.

I have also joined a world with a plain character, spawned a mustard, and checked the consumption time in the in-game UI. This seems to work!

### CODE REVIEW

**Comment by Photoloss** [comment]
Not that it matters much either way but is there an option for a manual override? Something like a worn drinking hat or the stimulant injector might warrant an active usage time much lower than 1s because it can operate in parallel with most other player actions.

**Comment by PatrikLundell** [comment]
The check for no mutations is a bit odd: your mustard guzzling normie (well, it's still an oddball) would then suddenly become an instant guzzler when gaining e.g, fur.

**Comment by Photoloss** [comment]
> The check for no mutations is a bit odd: your mustard guzzling normie (well, it's still an oddball) would then suddenly become an instant guzzler when gaining e.g, fur.

Based on the code changes this seems to be a phrasing issue. The minimum base consumption time still applies, but explicit mutation modifiers can reduce this value afterwards. So with Wide Mouth you would guzzle mustard at 0.25s per gulp.

**Comment by Gooseheaded** [comment]
Do we think we should increase the precision of this logic, then? To clearly support scenarios where consuming would take eg. 0.25s?

**Comment by Photoloss** [comment]
> Do we think we should increase the precision of this logic, then? To clearly support scenarios where consuming would take eg. 0.25s?

Since items should be able to specify arbitrary (usually higher) usage times anyway it should be doable with a simple item flag "INSTANT_CONSUME_OK" or whatever, similar to how we have ["ZERO_WEIGHT" on the battery charge "item"](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/items/ammo.json#L12). The only snag is that the flag should also be queried for certain containers to be useful (e.g. the worn jet injector or drinking hat) but requiring "ANY container has flag" would let you instantly drink from the hat even if it's stashed inside a cardboard box in your backpack.

Based on my reasoning for even considering such an option checking for "comestible has flag OR (container has flag AND is equipped)" should cover everything. It won't matter much in practice though so if you can't get it working easily the current changes should still be fine.

**Comment by I-am-Erk** [comment]
For the purposes of this PR, I am fine with merging a flat 1s minimum, but I have no opposition to an "INSTANT_CONSUME_OK" type flag, either in this or another - provided there is an existing use case that can take it as a demo. We're not usually in the habit of having item flags that aren't used, because they get forgotten and become dead code.

Option 2 would be a "minimum_consume_time" json attribute defaulting to "1 s" if not specified.

Neither of these need to be done on a first implementation, there aren't many cases where instant consumption is desirable to the point that 1s consume times are too long, I believe.

**Comment by Photoloss** [comment]
TL;DR: probably better to merge without further additions unless it's a trivial effort.

> Option 2 would be a "minimum_consume_time" json attribute defaulting to "1 s" if not specified.

I definitely can't think of any vanilla item justifying that addition right now. Turns out most "assisting devices" have explicitly defined usage actions whose time customisation is a separate issue. Definitely reads like a neater way of handling things if it doesn't take a lot more work to implement though. The drinking hat realistically should work more like the headlamp so it'd use "transformation" and "consumption *rate*" values instead.

**Comment by kevingranade** [comment]
Your unit test needs a pass through astyle, or you can look at the bottom of the "display corrections" section of the astyle test results to see what needs to be changed.
